packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -438,11 +438,36 @@ export const run = async () => {
       try {
         await func()
       } catch (err) {
-        logger.error('Error submitting batch', {
-          message: err.toString(),
-          stack: err.stack,
-          code: err.code,
-        })
+        switch (err.code) {
+          case 'SERVER_ERROR':
+            logger.error(`Encountered server error with status ${err.status}`, {
+              message: err.toString(),
+              stack: err.stack,
+              code: err.code,
+            })
+            break
+          case 'NETWORK_ERROR':
+            logger.error('Could not detect network', {
+              message: err.toString(),
+              stack: err.stack,
+              code: err.code,
+            })
+            break
+          case 'UNPREDICTABLE_GAS_LIMIT':
+            logger.error('Cannot estimate gas, transaction likely reverted', {
+              message: err.toString(),
+              stack: err.stack,
+              code: err.code,
+            })
```

Comment:
In this case I think you can strip out the revert reason and make that the `message`

Reply:
Ideally we have unique errors per revert reason